__Задача 1__

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите `M(X)`, `D(X)`.

__Решение__

Пусть Х - дискретная случайная величина количества угаданных билетов. Вероятность угадать вопрос $\dfrac{1}{4}$. Колчество билетов 5. Расчитаем по формуле Бернулли вероятность P(X=0), P(X=1), P(X=2), P(X=3), P(X=4),P(X=5)


In [80]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt

In [81]:
def bernulli(n,k,p):
    from math import factorial
    C=factorial(n)/(factorial(k)*factorial(n-k))
    return C*(p**k)*(1-p)**(n-k)

In [82]:
p=0.25
n=5
R=[]
for i in range(n+1):
    R.append(bernulli(n,i,p))
    R[i]=round(R[i],7)

In [83]:
columns=['P(X=0)', 'P(X=1)', 'P(X=2)', 'P(X=3)', 'P(X=4)','P(X=5)']
DT=dict.fromkeys(columns)
count=0
for value in DT:
    DT[value]=R[count]
    count+=1

__Ответ__ 

Закон распределения числа правильных ответов при простом угадывании:

In [84]:
pd.DataFrame(DT, index=[0])

P(X=0)    P(X=1)    P(X=2)    P(X=3)    P(X=4)    P(X=5)
0  0.237305  0.395508  0.263672  0.087891  0.014648  0.000977

In [8]:
np.array(R).sum()

1.0

Найдем M(X) и D(X)

In [9]:
M=n*p
M

1.25

In [10]:
D=n*p*(1-p)
D

0.9375

__Задача 2__

Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

__Решение__

Пусть X дискретная случайная величина. Вероятность попадания в цель р=0.01. Количество выстрелов n=200. Число попаданий k1=5 и k2=10.
Создаем Массив Х в котором рассчитываем поформуле бкернулли вероятность  для каждого числа выстрелов

In [14]:
p=0.01
n=200
k1=5
k2=10
X=[]
for i in range(k1,k2+1):
    X.append(bernulli(n,i,p))
X

[0.03572335672160829,
 0.011727364580325954,
 0.0032829851783307863,
 0.0008000203780528306,
 0.000172394963620812,
 3.326003843593443e-05]

__Ответ__

Сумма вероятностей количества попаданий от 5 до 10 и есть вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10 

In [18]:
np.array(X).sum()

0.051739381860374606

__Задача 3__

В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

_Подсказка_. Как посчитать `P(X >= k)` для величины `X`, имеющей распределение Пуассона? Здесь предстоит немножко покодить.

__Решение__

Подсчитываем вероятности 10 попаданий при n выстрелах и складываем их до тех пор, пока суммарная вероятность не будет равна или больше 0.9 
Находим решение через формулу Бернулли:

In [75]:
p=0.01
P=0
k=10
n=10
while P<0.9:
    P+=bernulli(n,k,p)
    n+=1
n, P

(473, 0.9067353190384637)

Находим решение через формулу Бернулли. При этом интенсивность $а=n\cdot p$, и меняется в зависимости от количества выстрелов:

In [36]:
from math import factorial
def puasson(n,k,p):
    return ((n*p)**k)*(np.exp(-n*p))/factorial(k)

In [76]:
p=0.01
k=10
n=10
P=0
while P<0.9:
    P+=puasson(n,k,p)
    n+=1
n, P

(471, 0.9088382632384054)

__Ответ__

По формуле Бернулли n=473

По формуле Пуассона n=471

__Решение после того, как посмотрел третий урок:__

In [92]:
def puasson_le(n,k,p):
    return sum(puasson(n,k0,p) for k0 in range(k))

In [93]:
def puasson_geq(n,k,p):
    return 1-puasson_le(n,k,p)

In [100]:
p=0.01
k=10
param_n = np.arange(100, 20000)

for n in param_n:
    proba = puasson_geq(n,k,p)
    
    if n % 200 == 0:
        print(f'{n}\t{proba}')
    
    if proba >= 0.9:
        break
    
print(f'n = {n}')
print(f'proba = {proba}')

200	4.6498075017109564e-05
400	0.00813224279693403
600	0.08392401699487573
800	0.28337574127298915
1000	0.5420702855281477
1200	0.7576078383294876
1400	0.890600630357261
n = 1421
proba = 0.9001760722984115
